aperturedb/CSVParser.py

```diff
@@ -86,43 +88,82 @@ def _parse_prop(self, key, val=None):
             val = {"_date": val}
         return key, val
 
-    def parse_properties(self, idx):
+    def parse_properties(self, df, idx):
 
         properties = {}
-        if self.props_keys:
+        if len(self.props_keys) > 0:
             for key in self.props_keys:
-                prop, value = self._parse_prop(key, self.df.loc[idx, key])
-                if value == value:  # skips nan values
-                    properties[prop] = value
+                # Handle Date data type
+                if key.startswith("date:"):
```

Comment:
_parse_prop already takes care of this . This is redundant. (or a merge problem from most recent develop)